tests/jetstream.rs

```diff
@@ -165,3 +167,45 @@ fn jetstream_libdoc_test() {
     let flipped: std::io::Result<Vec<usize>> = results.into_iter().collect();
     let _sizes: Vec<usize> = flipped.unwrap();
 }
+
+#[test]
+fn jetstream_idle_heartbeat() {
+    let (_s, nc) = run_basic_jetstream();
+
+    let stream = nc
+        .create_stream(StreamConfig {
+            name: "stream".to_string(),
+            ..Default::default()
+        })
+        .unwrap();
+
+    let mut consumer = nc
+        .create_consumer(
+            "stream",
+            &ConsumerConfig {
+                deliver_subject: Some("subject".to_string()),
+                durable_name: Some("consumer".to_string()),
+                idle_heartbeat: 200000000,
+                ..Default::default()
+            },
+        )
+        .unwrap();
+
+    consumer
+        .process(|message| {
```

Comment:
We want these HBs handled under the covers like the Go client and not exposed to the application unless the HB detects that we missed messages etc.